manticore/platforms/linux.py

```diff
@@ -2909,13 +2909,15 @@ def execute(self):
                 self.check_timers()
                 self.sched()
         except (Interruption, Syscall) as e:
+            index: int = self._syscall_abi.syscall_number()
+            self._syscall_abi._cpu._publish("will_invoke_syscall", index)
             try:
                 self.syscall()
                 if hasattr(e, "on_handled"):
                     e.on_handled()
             except RestartSyscall:
                 pass
-
+            self._syscall_abi._cpu._publish("did_invoke_syscall", index)
```

Comment:
This `RestartSyscall` makes me think the `did_invoke_syscall` should appear within the `try` block, but I forget when `RestartSyscall` is raised and what the state of Manticore is at that point. What are your thoughts?

Reply:
I think you're right. From what I can see in Pycharm it looks like `RestartSyscall` is for `sys_write` [here](https://github.com/trailofbits/manticore/blob/5a258f499098394c0af25e2e3f00b1b603c2334d/manticore/platforms/linux.py#L1792). I will move this 🙂